dotnet-authserver/src/TeacherIdentity.AuthServer/Oidc/TeacherIdentityApplicationManager.cs

```diff
@@ -39,6 +40,7 @@ public override async ValueTask PopulateAsync(OpenIddictApplicationDescriptor de
         {
             teacherIdentityApplicationDescriptor.ServiceUrl = await Store.GetServiceUrlAsync(application);
             teacherIdentityApplicationDescriptor.TrnRequirementType = await Store.GetTrnRequirementTypeAsync(application);
+            await Store.SetTrnMatchPolicy(application, teacherIdentityApplicationDescriptor.TrnMatchPolicy);
```

Comment:
```suggestion
            teacherIdentityApplicationDescriptor.TrnMatchPolicy = await Store.GetTrnMatchPolicy(application);
```

Reply:
Added new method as suggested and I have also updated the method names with Async prefix to match the rest